### PR TITLE
Reworks xenomorphs

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Reworks xenomorphs
Xenomorph disarm now takes 1.6 seconds between each strike (You can still get grouped up on)
Alien hunters are slower than they previously were (.5 movement decrease vs 1)
Alien hunter's leap now only stuns them for 3 seconds on miss, has a 3 second cooldown between leaps. immobilizes you while leaping, moves a bit faster, and can be used in zero G
Alien hunters no longer suffer from grab slowdown
Alien hunters have higher HP (125 -> 150)
Alien sentinels have higher HP (150 -> 175)
Alien drones now have higher HP (100 -> 150)
Alien queens now have higher HP (250 -> 300)
Alien queens now have a "Tail Lash" action, knockdown down everyone around them for a few seconds, 10 second cooldown
Drone and Queen plasma vessels have had their max amount of plasma dropped to 300 from 500
Hunter plasma vessels have their max plasma dropped from 150 to 100
Alien larva plasma vessels have their max plasma dropped from 100 to 50
Neurotoxin now stuns borgs for 10 seconds and has a lower cooldown of 3 seconds
Alien weeds take 2 seconds to be planted now along with alien eggs
Alien resin structures now take 3 seconds to be planted
Resin doors have had their HP decreased from 100 to 60, now cost the same as other structures
Adds a new resin structure, the alien rejuvination pod. Heals aliens a bit faster than weeds do and slowly cures viruses, has dick all integrity, doesn't keep people buckled in it for long, and slowly revives dead aliens in it. Retrieving your dead and keeping people alive is a much higher priority now.
Xenos can be infected with viruses again
Transfer plasma now can be used on any xeno on screen, hunters and larva get transfer plasma now
Corrosive acid now has a 15 second cooldown between uses
You can now hijack xeno organs using an Hemostat
All alien organs are now in the chest
Alien organs can be sold at cargo for a pretty penny. 
When Hijacked, xenomorph organs have all new effects instead of the spell that just don't work against xenos.
- Plasmavessel: Syphon plasma
Click on a xeno on your screen to instantly syphon 150 plasma, 10 second cooldown
- Acid Gland: Blazing touch
Click on a resin object in melee range to instantly melt it, Click on a living mob in melee range to deal 60 fire damage and set it ablaze, 15 second cooldown with 100 plasma cost
- Neurotoxin Gland: Neurotoxin spit (human version)
Fires neurotoxin that stuns xenos on hit for 10 seconds. same cost but a slightly shorter delay of 2 seconds
- resin spinner: consume resin
Allows you to destroy weeds or resin in melee range, no cooldown with destroying weeds and costs 10 plasma to destroy a structure, 5 second cooldown
- Egg Sac: Combust facehuggers and eggs
Allows you to detonate facehuggers and eggs, they'll glow red and explode in a small explosion after 3 seconds, plasma cost of 25 and a 5 second cooldown.
Acid armor has been removed from mechs, acid can now be removed from objects using cleaning
Hulk has been nerfed against xenos

Stuff I'm rather tempted to do:
Make xenos respect knockdown and stamina to an extent. They won't go into stamcrit but they'll get slowed from it (like stuns did in the past), knockdown will be less severe but it would let oldcrit be a bit less annoying with them along with a lower "extra" healthbar from crit, along with leap being able to be made less punishing
Allow facehuggers to assume multiple forms by a drone. hell might just let drones make them sentient as well
Make hunters loud when moving on surfaces that aren't resin. 
Qwerty mentioned xeno blood actually doing something like being acidic, I think that's based but it could also be just annoying.

## Why It's Good For The Game
Right now xenos are in a pretty weird state, they're fine balance wise but kinda suck to play and play against. They're not tanky at ALL so they end up dying to random stuff, the fact they don't have a baton style cooldown on disarm means that often times you'll have a ping war with someone and they either grab their weapon and shoot ya or you disarm them down and they're just dead. There's also very few differences between each cast, sentinels have an underwhelming projective, drones are hunters but slower, and hunters are just better. This PR seeks to try and give actual roles to each cast.
- Hunters
Focused on taking down unarmed targets and moving aliens/bodies to/from the front lines of combat back to the base, their leap is great for crossing people up too
- Sentinels
Focuses on 1 on 1 combat, their neurotoxin is a great opener which can be spammed down a choke if need be. They can take more punishment than other aliens meaning these are the guys you want to generally be fighting.
- Drones (Still could probably use a bit more to make em different)
A support role focusing on creating a hive to protect larva, the infected, and the dead who are being slowly revived.
- Queen
You are the most efficient way to create eggs by far. With high HP and strong attacks, your job is to serve as an ace in the hole to deal with threats other xenos would struggle with. Be warned however, your death and capture can easily spell doom for a hive if they obtain your egg sack and can't protect from an incoming assault.

If you do manage to kill a xeno as crew, it really doesn't feel impactful as the organs do absolutely nothing except for the hive and plasma vessel. This just gives a lot more to chew on and makes losing a lot of xenos actually hurt. 
This is also pretty straightforward but xenos relying on very hard counters ain't fun to use or to play against.

## Testing
So much clicking on spells....

## Changelog
:cl:
tweak: Aliens have been reworked
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
> Xenomorph disarm no longer disarms them of their active item (I don't think people want to play ping wars anymore)

You would then be able to go prone to fight Xenos with melee (welders) and never get disarmed. This alone is a straight up no from me. Axe this or I cant in good conscious vote for this PR as a whole no matter what the other changes are.

Not a fan of dropping the Queen's plasma to 300. Drones are w/e.

Not a fan of the timer on egg laying - the "fun" part of Queen gameplay is base making and coordinating attacks, this is just tedium that very rarely matters.

The entire implants section is kinda wack and i dont really think it makes sense. Why would organs suddenly give you powers that Xenos themselves dont have? Unless there's more logic applied to these I'm against them. Have it be some weird Bio 8 machine or anything that makes more sense than "here have some MMO powers to fight benos".

I do think the Rejuv Sac is a neat idea. Feels very xeno to me.

Other changes are so-so/okay.

Overall, gonna go with a "**No**" vote as currently implemented ([7318e4a](https://github.com/ParadiseSS13/Paradise/pull/23054/commits/7318e4a50089e456642d5af63b949f8504f25b6c)).

**Comment by matttheficus** [comment]
Also, this would mean 6 seconds to stun a single person as a Xeno.

That's a bit insane and also needs to be addressed if an approval is wanted.

**Comment by Rurik123** [comment]
So if I've read correctly: You are nerfing both hunter speed and time to stamcrit, without any compensating buff to health or removal of the burn modifier? 

The reason hunters are fast and stamcrit fast is because, when accounting for their low HP with burn modifier, they die fast. You cannot touch one without touching the other. You have the right idea with the leap change, but it is not enough to account for both of the above. 

Hard no until the above is adequately addressed.

The MMO organs and rejuv sac is... odd, but I see the minigame you're trying to create (capture the flag, except its the dead alien on the floor) and I don't feel strongly against nor for it. At the very least its limited to the xeno dynamic, and wont bleed into other antag fights. 

Shifting the queen into a combat focus as you're doing is fine.

> Xenos can be infected with viruses again

Its a fucking bioweapon not an elderly patient at the the local ward. I'm aware you made it curable in the pod, but that doesn't change that Xeno sneeze spam is annoyance with 0 benefit.

**Comment by spacegeese** [comment]
the days of accidentally causing a xeno infestation with an adminspawn eggsac are over xenolads

**Comment by Fordoxia** [comment]
>Neurotoxin now stuns borgs for 10 seconds and has a lower cooldown of 2 seconds

Bruh they already get instantly put to sleep on melee.

File: code/_onclick/hud/alien_hud.dm [hunk 1]
```diff
@@ -18,6 +18,19 @@
 	var/mob/living/carbon/alien/humanoid/A = usr
 	A.nightvisiontoggle()
 
+/mob/living/carbon/alien/proc/nightvisiontoggle()
+	if(!nightvision)
+		see_in_dark = 8
+		lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+		nightvision = TRUE
+		hud_used.nightvisionicon.icon_state = "nightvision1"
+	else if(nightvision)
```
Comment (Henri215) [inline code/_onclick/hud/alien_hud.dm:27]: This else if is necessary?
Could snake_case the proc name too.

File: code/datums/spells/alien_spells/build_resin_structure.dm [hunk 2]
```diff
@@ -27,15 +31,54 @@
...
+		return
+	if(!plasma_check(25, user))
+		to_chat(user, "<span class='noticealien'>You don't have enough plasma to perform this action!</span>")
+		return
+	var/static/list/resin_objects = list(/obj/structure/alien/resin, /obj/structure/alien/egg, /obj/structure/bed/nest, /obj/structure/bed/revival_nest)
+	for(var/resin_type in resin_objects)
+		if(!istype(target, resin_type))
+			to_chat(user, "type to check: [resin_type]")
+			continue
+			to_chat(user, "Got past the istype checks")
```
Comment (Henri215) [inline code/datums/spells/alien_spells/build_resin_structure.dm:77]: Forgot to change these?

File: code/datums/spells/alien_spells/transfer_plasma.dm [hunk 3]
```diff
@@ -1,28 +1,55 @@
...
+
+	user.Beam(target, icon_state = "sendbeam", time = 2 SECONDS, beam_color = "#f180bd")
+	target.add_plasma(50)
+	to_chat(user, "<span class='noticealien'>You have transfered 50 plasma to [target].</span>")
+	to_chat(target, "<span class='noticealien'>[user] has transfered 50 plasma to you!</span>")
+	return TRUE
+
+/obj/effect/proc_holder/spell/alien_spell/syphon_plasma
+	name = "Syphon Plasma"
+	desc = "Syphons 50 plasma from a nearby alien"
```
Comment (Henri215) [inline code/datums/spells/alien_spells/transfer_plasma.dm:25]:
```suggestion
	name = "Syphon plasma"
	desc = "Syphons 50 plasma from a nearby alien."
```

File: code/game/objects/structures/stool_bed_chair_nest/alien_nests.dm [hunk 4]
```diff
@@ -110,3 +110,124 @@
 
 /obj/structure/bed/nest/prevents_buckled_mobs_attacking()
 	return TRUE
+
+/obj/structure/bed/revival_nest
+	name = "alien rejuvination nest"
```
Comment (Henri215) [inline code/game/objects/structures/stool_bed_chair_nest/alien_nests.dm:115]:
```suggestion
	name = "alien rejuvenation nest"
```

File: code/modules/mob/living/carbon/alien/alien_death.dm [hunk 5]
```diff
@@ -12,7 +12,7 @@
 	animation.icon = 'icons/mob/mob.dmi'
 	animation.master = src
 
-	playsound(src.loc, 'sound/goonstation/effects/gib.ogg', 50, 1)
+	playsound(loc, 'sound/goonstation/effects/gib.ogg', 50, 1)
```
Comment (Henri215) [inline code/modules/mob/living/carbon/alien/alien_death.dm:15]:
```suggestion
	playsound(loc, 'sound/goonstation/effects/gib.ogg', 50, TRUE)
```

File: code/modules/mob/living/carbon/alien/larva/larva_inventory.dm [hunk 6]
```diff
@@ -1,3 +1,3 @@
-//can't unequip since it can't equip anything
+//can't unequip since it can't equip anything // why the fuck is this it's own file
```
Comment (Henri215) [inline code/modules/mob/living/carbon/alien/larva/larva_inventory.dm:1]: 😆

File: code/modules/surgery/organs/subtypes/xenos.dm [hunk 7]
```diff
@@ -2,59 +2,98 @@
 	origin_tech = "biotech=5"
 	icon_state = "xgibmid2"
 	var/list/alien_powers = list()
+	var/list/human_powers = list()
 	tough = TRUE
 	sterile = TRUE
+	/// Amount of credits that will be recived by selling this in the cargo shuttle
```
Comment (Henri215) [inline code/modules/surgery/organs/subtypes/xenos.dm:8]:
```suggestion
	/// Amount of credits that will be received by selling this in the cargo shuttle
```

**Review by Henri215 — COMMENTED** [review]
Looking at icondiff i imagine its showing a bunch of stuff you didnt touch, right?

**Review by S34NW — CHANGES_REQUESTED** [review]
3 weeks without changes, please deconflict and update.

File: code/game/objects/structures/stool_bed_chair_nest/alien_nests.dm [hunk 8]
```diff
@@ -110,3 +110,124 @@
...
+		dust_if_respawnable(C)
+		dead_alien.mind.name = dead_alien.name
+		dead_alien.mind.assigned_role = SPECIAL_ROLE_XENOMORPH
+		dead_alien.mind.special_role = SPECIAL_ROLE_XENOMORPH
+		SEND_SOUND(dead_alien, sound('sound/voice/hiss5.ogg'))
+
+/obj/structure/bed/revival_nest/proc/decay_dead_carbon(mob/living/carbon/human/dead_carbon)
+	if(HAS_TRAIT(dead_carbon, TRAIT_SKELETONIZED))
+		return
+	if(ismachineperson(dead_carbon) || isrobot(dead_carbon)) // A potential idea is to make these function like working joes in isolation, but I don't have sprites for em
```
[the remark this answers is not in the document]
Comment (GDNgit) [reply code/game/objects/structures/stool_bed_chair_nest/alien_nests.dm:202]: MOB_ROBOTIC isn't used anymore and isn't frequently updated, I think the istype checks are alright

[the remark this answers is not in the document]
Comment (GDNgit) [reply code/game/objects/structures/stool_bed_chair_nest/alien_nests.dm:202]: Actually screw it let's start using them now

File: code/datums/spells/alien_spells/build_resin_structure.dm (hunk 2 above)
Comment (GDNgit) [reply code/datums/spells/alien_spells/build_resin_structure.dm]: reviews addressed

**Comment by matttheficus** [comment]
Spamming leap looks REALLY silly/buggy if you do it down long hallways.

**Comment by spacegeese** [comment]
>Eggmorphs dead carbons in it

what does this mean? like it makes dead carbons into hugger eggs or it makes them into larva like cm does?

**Comment by spacegeese** [comment]
>Eggmorphs dead carbons in it

what does this mean? like it makes dead carbons into hugger eggs or it makes them into larva like cm does?

**Comment by silverplatedelta** [comment]
Given I've read this all over and had a chance to play with it, I have some significant concerns. In my personal opinion, the issue with xenos as an antag is the stamina and stun based combat, where a loss means you are pulled into the abyss and removed from the round. Being hit four times then yoinked into the darkness isn't fun to fight against, especially when your attacker is moving at methspeed. 

After this rework, you are now hit with a ranged attack, hit two more times, then pulled into the darkness and removed from the round, with your attackers taking on average another three hits to down. Add in the ability for xenos to revive their dead, a situation in which crew numbers rapidly dwindle while xenos stay at least the same is likely (at least until crew gets some gamer gear). 

These changes have failed to address, in my opinion, the actual issue with xenos: you are hit 3-4 times, pulled into the darkness and removed from the round. This deprives several departments of a gameplay loop, mostly medical here, as they don't get patients until they have literally 40 from the nest being busted, and by then the situation is pretty much sorted. This is a stark contract from something like spiders, and blob, which use mechanics (or the lack thereof) to re-introduce crewmembers into the round to get back in the battle.

I will note, the idea of seizing xeno bodies, removing and hijacking organs to use against them is neat, but I'm unsure if its going to be used often. Being locked behind biotech six is a decently large hurdle especially earlier in the round, and puts these abilities all locked behind botany doing their job (honestly for this, just know its not the most common thing).

The meta I feel this encourages is:
Xenos: Operate in groups of 2-3, take down crew groups of the same size, run them to base, repeat. Spam spit down chokes and run in and grab who you can, given you are tanky enough.

Crew: Getting in close contact is a bad idea. the ideal move is liberal use of fire, fire chemicals, explosives, and spontaneous combustion viruses (apparently we are doing this again, hotpox is BACK baby). Avoid line of sight. Given the removal of acid armor from mechs, as well as there being no real way to combat acid placement, they are significantly reduced in use. Just sit back on a sight line and pew down a hallway.

**Review by S34NW — APPROVED** [review]
Four balance members in favour, a while without major reported issues on TM, it's time to go son.

**Pull request merged**